test/solc/CommandLineParser.cpp

```diff
@@ -224,6 +224,15 @@ BOOST_AUTO_TEST_CASE(cli_mode_options)
 	}
 }
 
+BOOST_AUTO_TEST_CASE(no_append_metadata)
+{
+	vector<string> commandLine = {"solc", "--no-append-metadata", "contract.sol"};
+	CommandLineOptions parsedOptions = parseCommandLine(commandLine);
+	bool assert = parsedOptions.metadata.format == CompilerStack::MetadataFormat::NoMetadata;
+
+	BOOST_TEST(assert);
+}
```

Comment:
This test is fine but what I meant in https://github.com/ethereum/solidity/pull/13265#discussion_r938088556 was that this file has tests for every input mode, listing all available options. And also a (still incomplete) test for options that are not allowed. Could you add the option to test for the compiler input mode?

Reply:
Just rebased and fixed the comment on `metadata.rst`. This is what's left now. Will try.

Reply:
@cameel Is something like adding a `for (InputMode inputMode: {InputMode::Compiler, InputMode::CompilerWithASTImport}` etc what you are looking?

Maybe failures for `InputMode::Assembler` and `Linker`?

Reply:
> Is something like adding a `for (InputMode inputMode: {InputMode::Compiler, InputMode::CompilerWithASTImport})` etc what you are looking?

Kinda, but not in a new test. Rather just add it in `cli_mode_options`.

> Maybe failures for `InputMode::Assembler` and `Linker`?

And for that we have `invalid_options_input_modes_combinations`.

Reply:
> Rather just add it in cli_mode_options.

This is a little bit annoying because you can't have `--no-cbor-metadata` and `--metadata-hash=swarm` together. Which breaks the stuff.

Reply:
Ah, ok then. In that case a separate test makes sense.